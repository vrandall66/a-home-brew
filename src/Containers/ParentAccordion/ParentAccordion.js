import React, { Component } from 'react';
import { IoIosArrowDropdown, IoIosArrowDropdownCircle } from 'react-icons/io';
import { PropTypes } from 'prop-types';
import './ParentAccordion.scss';

export class ParentAccordion extends Component {
  constructor() {
    super();
    this.state = {
      isExpanded: false
    };
  }

  handleExpand = () => {
    this.setState({ isExpanded: !this.state.isExpanded });
  };

  render() {
    const { title, accordions } = this.props;
    const { isExpanded } = this.state;
    const ingredients = accordions.map((accordion, index) => {
      return <li key={index}>{accordion}</li>;
    });
    return (
      <div className='Accordion__section'>
        <button className='Accordion' onClick={this.handleExpand}>
          <p className='Accordion__title'>{title}</p>
          {isExpanded ? (
            <IoIosArrowDropdownCircle className='Accordion__icon' />
          ) : (
            <IoIosArrowDropdown className='Accordion__icon' />
          )}
        </button>
        {isExpanded && (
          <div className='Accordion__content'>
            <ul>{ingredients}</ul>
          </div>
        )}
      </div>
    );
  }
}

export default ParentAccordion;

ParentAccordion.propTypes = {
  title: PropTypes.string,
  accordions: PropTypes.array
};
