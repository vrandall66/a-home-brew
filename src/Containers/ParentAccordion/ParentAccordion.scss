.Accordion {
  width: 100%;
}

.Accordion__section {
  width: 100%;
}