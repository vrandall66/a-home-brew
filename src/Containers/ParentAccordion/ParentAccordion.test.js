import React from 'react';
import { shallow } from 'enzyme';
import ParentAccordion from './ParentAccordion';

describe('ParentAccordion', () => {
  let wrapper;
  let accordions = [{}, {}, {}];

  beforeEach(() => {
    wrapper = shallow(<ParentAccordion accordions={accordions} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
