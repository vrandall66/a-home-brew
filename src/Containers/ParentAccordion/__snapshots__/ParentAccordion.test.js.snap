// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentAccordion should match the snapshot 1`] = `
<div
  className="Accordion__section"
>
  <button
    className="Accordion"
    onClick={[Function]}
  >
    <p
      className="Accordion__title"
    />
    <IoIosArrowDropdown
      className="Accordion__icon"
    />
  </button>
</div>
`;
